src/components/Data/Layout.vue

```diff
@@ -21,17 +23,39 @@
   import {LIST_INDEXES_AND_COLLECTION} from '../../vuex/modules/index/mutation-types'
   import {FETCH_COLLECTION_DETAIL} from '../../vuex/modules/collection/mutation-types'
   import Treeview from './Leftnav/Treeview'
+  import NotFound from '../404'
   import {SET_TOAST} from '../../vuex/modules/common/toaster/mutation-types'
 
   export default {
     name: 'DataLayout',
     components: {
-      Treeview
+      Treeview,
+      NotFound
+    },
+    data () {
+      return {
+        routeExist: true
+      }
+    },
+    methods: {
+      setRouteExist () {
+        this.routeExist = true
+        let index = this.$store.state.route.params.index
+        if (typeof index !== 'undefined' && this.$store.state.index.indexes.indexOf(index) === -1) {
+          this.routeExist = false
+        } else {
+          let collection = this.$store.state.route.params.collection
```

Comment:
prefer `const` over `let` for constants

Reply:
You may also use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to initialize both `index` and `collection` in one command, like so:

`const {index, collection} = this.$store.state.route.params;`